src/Compilers/CSharp/Portable/Symbols/Source/SourceOrdinaryMethodSymbol.cs

```diff
@@ -401,6 +401,7 @@ protected sealed override SourceMemberMethodSymbol BoundAttributesSource
 
         internal sealed override OneOrMany<SyntaxList<AttributeListSyntax>> GetAttributeDeclarations()
         {
+            Debug.Assert(PartialDefinitionPart is null);
```

Comment:
I didn't follow this assertion. Could you explain?

Reply:
It's not expected to call this on an implementation part symbol, because the definition part "owns" binding the attributes on it.

Reply:
I think I understand: `PartialDefinitionPart` returns null if we're not dealing with a partial, or if this is the definition part. The assertion says we don't expect to call this method on the definition part.
If that's correct, consider leaving a comment (and similar in property analog code)

Reply:
Yes, and importantly, the check is on the PartialDefinitionPart property, and not IsPartialImplementation, because we could still call this on an implementation part symbol when the definition part is missing, for example.

Reply:
> PartialDefinitionPart returns null if we're not dealing with a partial, or if this is the definition part

Yes, and these are the scenarios where the assertion passes.

> The assertion says we don't expect to call this method on the definition part.

Given the above I think you probably meant to say implementation part here, but it sounds like you understood what the assert is doing.

I'll add comments to these asserts.

Reply:
> Given the above I think you probably meant to say implementation part here

Yes. Thanks for clarifying.

I'm still missing something. How come we can have this kind of assertion in `SourceOrdinaryMethodSymbol.GetAttributeDeclarations()` and `SourcePropertySymbol.AttributeDeclarationLists`, but we can't have it in `SourcePropertyAccessorSymbol.GetAttributeDeclarations()`?

Reply:
`SourcePropertyAccessorSymbol.GetAttributeDeclarations()` is special because accessors can have `[param: ...]` targeted attribute lists. Partial parameter attribute handling is weird, and prefers getting attributes through the implementation part and ordering the implementation attributes first.

Normally when binding the attributes on an accessor, we will get the attribute syntax lists through the definition symbol, and order them definition->impl.

But, for `[param: ...]` attributes on a set accessor, we will actually get the attribute lists through the impl symbol, and order them impl->definition.

I considered different ways of factoring this in the symbol APIs, and even considered making a behavior change to make parameters consistent with everything else, but ended up finding this the most expedient. Actually, this is part of what was frying my brain yesterday.